dist/gcc-compatible/EverCrypt_HMAC.c

```diff
@@ -571,51 +566,17 @@ EverCrypt_HMAC_compute_blake2s(
     opad[i] = xi ^ yi;
   }
   uint32_t s[16U] = { 0U };
-  uint32_t *r0 = s;
-  uint32_t *r1 = s + (uint32_t)4U;
-  uint32_t *r2 = s + (uint32_t)8U;
-  uint32_t *r3 = s + (uint32_t)12U;
-  uint32_t iv0 = Hacl_Impl_Blake2_Constants_ivTable_S[0U];
-  uint32_t iv1 = Hacl_Impl_Blake2_Constants_ivTable_S[1U];
-  uint32_t iv2 = Hacl_Impl_Blake2_Constants_ivTable_S[2U];
-  uint32_t iv3 = Hacl_Impl_Blake2_Constants_ivTable_S[3U];
-  uint32_t iv4 = Hacl_Impl_Blake2_Constants_ivTable_S[4U];
-  uint32_t iv5 = Hacl_Impl_Blake2_Constants_ivTable_S[5U];
-  uint32_t iv6 = Hacl_Impl_Blake2_Constants_ivTable_S[6U];
-  uint32_t iv7 = Hacl_Impl_Blake2_Constants_ivTable_S[7U];
-  r2[0U] = iv0;
-  r2[1U] = iv1;
-  r2[2U] = iv2;
-  r2[3U] = iv3;
-  r3[0U] = iv4;
-  r3[1U] = iv5;
-  r3[2U] = iv6;
-  r3[3U] = iv7;
-  uint32_t kk_shift_8 = (uint32_t)0U;
-  uint32_t iv0_ = iv0 ^ ((uint32_t)0x01010000U ^ (kk_shift_8 ^ (uint32_t)32U));
-  r0[0U] = iv0_;
-  r0[1U] = iv1;
-  r0[2U] = iv2;
-  r0[3U] = iv3;
-  r1[0U] = iv4;
-  r1[1U] = iv5;
-  r1[2U] = iv6;
-  r1[3U] = iv7;
-  uint64_t es = (uint64_t)0U;
-  K____uint32_t__uint64_t scrut0 = { .fst = s, .snd = es };
-  uint32_t *s0 = scrut0.fst;
+  Hacl_Blake2s_32_blake2s_init(s, (uint32_t)0U, (uint32_t)32U);
+  uint32_t *s0 = s;
   uint8_t *dst1 = ipad;
-  uint64_t ev0 = Hacl_Hash_Core_Blake2_init_blake2s_32(s0);
-  uint64_t ev10;
+  Hacl_Blake2s_32_blake2s_init(s0, (uint32_t)0U, (uint32_t)32U);
```

Comment:
there are two inits in a row here -- is this a bug in Hacl.HMAC? would be good to figure out why we re-initialize (or how to skip it, e.g. by revealing that alloca provides an already-initialized buffer or something)

Reply:
This is solved in #739 . Marking as resolved.